bot/bot.go

```diff
@@ -92,16 +92,27 @@ func (me *Bot) Monitor(trade config.Trade) error {
 
 // trade create an order to buy or sell.
 func (me *Bot) buy(trade config.Trade, price float64) error {
-	wallet, err := me.Binance.SymbolBalance(trade.BuyWith())
+	wallet, err := me.Binance.SymbolBalance(trade.GetSymbol())
 	if err != nil {
 		return err
 	}
 
-	if wallet == 0 {
+	// already bought.
+	if math.Floor(wallet) > 0 {
 		return nil
 	}
 
-	quantity := math.Floor(wallet / price)
+	wallet, err = me.Binance.SymbolBalance(trade.BuyWith())
+	if err != nil {
+		return err
+	}
+
+	// don't have enough to buy.
+	if wallet == 0 || wallet < trade.Limit {
+		return nil
+	}
+
+	quantity := math.Floor(trade.Limit / price)
 	if quantity == 0 {
 		return nil
```

Comment:
What do you think about log something whenever it returns `nil`?

Reply:
nice, this logs will give good visibility.

I added in [commit](https://github.com/rafaelbmateus/go-binance-bot/pull/3/commits/99ad641530c8844d7ad1b47a5c527a684d19ede7)